src/main/scala/akka/persistence/utils/HostAndPort.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package akka.persistence.utils
+
+/** Convenience class for holding host and port tuples and not having to depend on Guava.
+ *  @param host the hostname
+ *  @param portOpt the (optional) port
+ */
+case class HostAndPort private (host: String, portOpt: Option[Int]) {
+
+  // construction assertion
+  portOpt match {
+    case Some(p) => require(isValidPort(p))
+    case _       =>
+  }
```

Comment:
This assertion is not checked for sentinels configured with objects. I would prefer not to perform it here. I am also not totally convinced we need it. We do not perform any validation and checks on configuration, and this check create some inconsistency in the behavior.